src/ophys_etl/modules/segmentation/seed/seeder.py

```diff
@@ -0,0 +1,317 @@
+                            Seed(coordinates=inds,
+                                 value=block.max(),
+                                 excluded_by_roi=False,
+                                 excluded_by_quantile=False))
+                    col += self._seeder_grid_size
+                row += self._seeder_grid_size
+
+        seeds_sorted = sorted(seeds,
+                              key=lambda x: x['value'],
+                              reverse=True)
+        num_keep = int(self._keep_fraction * len(seeds_sorted))
+        self._candidate_seeds = seeds_sorted[:num_keep]
+        self._excluded_seeds = seeds_sorted[num_keep:]
+        for i in range(len(self._excluded_seeds)):
+            self._excluded_seeds[i]["exclusion_reason"] = \
+                    "excluded by fraction"
+
+    def log_to_h5_group(self,
+                        h5file: h5py.File,
+                        group_name: str = "seeding"):
+        """records some attributes of this seeder to an hdf5 group for
+        inspection purposes.
+
+        Parameters
+        ----------
+        h5file: h5py.File
+            an h5py.File object which has been opened in a writable mode
+        group_name: str
+            the name of the group to which this object's logged attrbiutes
+            will be inserted
+
+        """
+        # collect any candidates not even considered (perhaps a segmenter
+        # can impose a stopping condition different from running out of seeds
+        for seed in self._candidate_seeds:
+            seed["exclusion_reason"] = "never considered"
+            self._excluded_seeds.append(seed)
+
+        group = h5file.create_group(group_name)
+        group.create_dataset(
+                "provided_seeds",
+                data=np.array([i['coordinates']
+                               for i in self._provided_seeds]))
+        group.create_dataset(
+                "excluded_seeds",
+                data=np.array([i['coordinates']
+                               for i in self._excluded_seeds]))
+        group.create_dataset(
+                "exclusion_reason",
+                data=np.array([i['exclusion_reason'].encode("utf-8")
+                               for i in self._excluded_seeds]))
+        group.create_dataset(
+                "seed_image",
+                data=self._seed_image)
+
+
+class BatchImageMetricSeeder(ImageMetricSeeder):
+    """Can serve up to 'n_samples' size batches of seeds that exceed a
+    minimum distance criteria. This is useful for sending to parallel
+    jobs where the potential exclusion zones are unlikely reach the other
+    the other seeds.
+
+    Parameters
+    ----------
+    n_samples: int
+        the maximum number of seeds to provided per __next__() call.
+    minimum_distance: float
+        the minimum euclidean distance to allow in a single batch
+
+    """
+    def __init__(self, n_samples: int,
+                 minimum_distance: float, *args, **kwargs):
```

Comment:
Same concern about explicit vs implicity parameters.

Reply:
compromised by removing `*args` from passing through to base classes.